stdlib/array.ncl

```diff
@@ -45,6 +49,21 @@
         "%
       = fun l => %tail% l,
 
+    tail_but_reverse_bikeshed_me : forall a. Array a -> Array a
+      | doc "TODO"
+      # TODO: we need a slice primop
+      = fun l => l
+        |> reverse
+        |> tail
+        |> reverse,
```

Comment:
I proposed `cotail`, but people didn't like it :/

`pop` would be confusing because you don't know in an immutable language if it returns the element or the array.

Note that we can in fact make it `O(1)` in Nickel, but we would need a primop.

Another possibility is to have more explicit names like `without_last` and `without_first`. Less common than `tail` for the latter, but at least symmetric and might be more intuitive for non functional programmers.

Reply:
`pop_front` and `pop_back` would also be reasonable candidates.

Reply:
The newest option: `drop_first` and `drop_last`. Performance shouldn't really play into this decision since we're planning to add an array slice primop anyway. That will make both functions take constant time. Namingwise, `drop` makes it clear that the array is returned instead of the element and naming between `drop_first` and `drop_last` is uniform an unambiguous.